### ISSUE TITLE
Minetest - no fonts in launcher

### ISSUE THREAD

**Issue by kucharskim** [minetest/minetest#11349]
##### Minetest version
```
$ minetest --version
Minetest 5.1.1 (Linux)
Using Irrlicht 1.8.4
BUILD_TYPE=None
RUN_IN_PLACE=0
USE_GETTEXT=1
USE_SOUND=1
USE_CURL=1
USE_FREETYPE=1
USE_LUAJIT=1
STATIC_SHAREDIR="/usr/share/games/minetest"
```

##### OS / Hardware
Operating system: `Ubuntu 20.04.2 LTS`
CPU: `Intel(R) Celeron(R) CPU 3215U @ 1.70GHz`
GPU model: `Intel HD Graphics, driver=i915`
OpenGL version: `4.6 (Compatibility Profile) Mesa 20.2.6`
```
OpenGL vendor string: Intel
OpenGL renderer string: Mesa Intel(R) HD Graphics (BDW GT1)
OpenGL core profile version string: 4.6 (Core Profile) Mesa 20.2.6
OpenGL core profile shading language version string: 4.60
OpenGL core profile context flags: (none)
OpenGL core profile profile mask: core profile
```

##### Summary

Install with `apt install -y minetest
` and then start with `minetest`. In a launcher doesn't render any texts, like fonts are missing. I've seen this reported on Minetest forum here: <https://forum.minetest.net/viewtopic.php?t=23769> and screenshot in that post shows exactly the same problem. As described in that post I've worked around the problem via:

```
echo 'language = en' >> ~/.minetest/minetest.conf
```

However I'm wondering could that be fixed in the code, that no matter what is local language, fonts will display in the launcher anyway.

##### Steps to reproduce
I think setting up `locale` may be a contributing factor here:

```
$ cat /etc/default/locale 
#  File generated by update-locale
LANG="pl_PL.UTF-8"
```

```
$ locale
LANG=pl_PL.UTF-8
LANGUAGE=
LC_CTYPE="pl_PL.UTF-8"
LC_NUMERIC="pl_PL.UTF-8"
LC_TIME="pl_PL.UTF-8"
LC_COLLATE="pl_PL.UTF-8"
LC_MONETARY="pl_PL.UTF-8"
LC_MESSAGES="pl_PL.UTF-8"
LC_PAPER="pl_PL.UTF-8"
LC_NAME="pl_PL.UTF-8"
LC_ADDRESS="pl_PL.UTF-8"
LC_TELEPHONE="pl_PL.UTF-8"
LC_MEASUREMENT="pl_PL.UTF-8"
LC_IDENTIFICATION="pl_PL.UTF-8"
LC_ALL=
```

```
sudo apt install -y minetest
minetest
```

**Comment by kucharskim** [comment]
Link to screenshot in that post: <https://forum.minetest.net/download/file.php?id=20753>

**Comment by sfan5** [comment]
We don't support 5.1.1, try a newer version.

**Comment by kucharskim** [comment]
I see. For the record, at the time of this comment, latest release on GitHub is `5.4.1`.

**Comment by rubenwardy** [comment]
You can get up to date versions for Ubuntu using the Minetest official PPA: https://launchpad.net/~minetestdevs/+archive/ubuntu/stable

**Comment by neoh4x0r** [comment]
**TL;DR** The language must be set in minetest.conf (I don't see where it is detected)
I think the request was to autodetect the lanaguage without having to set it in minetest.conf

Moreover, this will have to be done even in 5.5.0

-----

see here: https://forum.minetest.net/viewtopic.php?p=312467#p312467

> The problem is not the language setting, but the FontDroidFallbakcFull font is missing in Ubuntu 17.10 and derivatives.
> 
> The easiest way is to add at the end of the minetest.conf file in the user's directory.
> 
> language = pl
> fallback_font_path = /usr/share/games/minetest/fonts/Cousine-Regular.ttf
> screenH = 1029
> screenW = 1920
> main_menu_technical_settings = true
> font_size = 14
> mono_font_size = 14
> fallback_font_size = 14
> 
> I chose this font because it is in the main game directory, but it may be different.
> 
> Settings can be changed after starting the game in Settings -> Advanced Settings, in the search field enter lang and correct entries

The reported issue in the forum was related to Minetest 0.4.16 from Kubuntu 17.10 

However, focal 20.04 has the DroidSansFallbackFull font present.

https://packages.ubuntu.com/focal/all/minetest-data/filelist
```
/usr/share/games/minetest/fonts/DroidSansFallbackFull-LICENSE.txt
/usr/share/games/minetest/fonts/DroidSansFallbackFull.ttf
```

**By default fallback_font_path is set to DroidSansFallbackFull.ttf**
```
std::string getDataPath(const char *subpath)
{
	return path_share + DIR_DELIM + subpath;
}

settings->setDefault("fallback_font_path", porting::getDataPath("fonts" DIR_DELIM "DroidSansFallbackFull.ttf"));

```

```
STATIC_SHAREDIR="/usr/share/games/minetest"
```

Therefore, since the fall-back font is present at **/usr/share/games/minetest/fonts/DroidSansFallbackFull.ttf**
and all default fonts in **src/defaultsettings.cpp** are present in the ubuntu minetest-data package, then
**The language must be set in minetest.conf**

@kucharskim have you changed any of the defaults related to font?

https://github.com/minetest/minetest/blob/edf098db637bf74d4675ce900dc9c0b8ee528a03/src/defaultsettings.cpp#L303
```
#if USE_FREETYPE
	settings->setDefault("freetype", "true");
	settings->setDefault("font_path", porting::getDataPath("fonts" DIR_DELIM "Arimo-Regular.ttf"));
	settings->setDefault("font_path_italic", porting::getDataPath("fonts" DIR_DELIM "Arimo-Italic.ttf"));
	settings->setDefault("font_path_bold", porting::getDataPath("fonts" DIR_DELIM "Arimo-Bold.ttf"));
	settings->setDefault("font_path_bold_italic", porting::getDataPath("fonts" DIR_DELIM "Arimo-BoldItalic.ttf"));
	settings->setDefault("font_bold", "false");
	settings->setDefault("font_italic", "false");
	settings->setDefault("font_shadow", "1");
	settings->setDefault("font_shadow_alpha", "127");
	settings->setDefault("mono_font_path", porting::getDataPath("fonts" DIR_DELIM "Cousine-Regular.ttf"));
	settings->setDefault("mono_font_path_italic", porting::getDataPath("fonts" DIR_DELIM "Cousine-Italic.ttf"));
	settings->setDefault("mono_font_path_bold", porting::getDataPath("fonts" DIR_DELIM "Cousine-Bold.ttf"));
	settings->setDefault("mono_font_path_bold_italic", porting::getDataPath("fonts" DIR_DELIM "Cousine-BoldItalic.ttf"));
	settings->setDefault("fallback_font_path", porting::getDataPath("fonts" DIR_DELIM "DroidSansFallbackFull.ttf"));

	std::string font_size_str = std::to_string(TTF_DEFAULT_FONT_SIZE);
#else
	settings->setDefault("freetype", "false");
	settings->setDefault("font_path", porting::getDataPath("fonts" DIR_DELIM "mono_dejavu_sans"));
	settings->setDefault("mono_font_path", porting::getDataPath("fonts" DIR_DELIM "mono_dejavu_sans"));

	std::string font_size_str = std::to_string(DEFAULT_FONT_SIZE);
#endif
```

PS: All of the above is present even in the latest versions of minetest (at the time of writing was 5.5.0)

**Comment by kucharskim** [comment]
@neoh4x0r, the only thing I've changed, after seeing no fonts and a bit of Google is this:

      echo 'language = en' >> ~/.minetest/minetest.conf

I didn't need to modify any font settings. I can confirm that `DroidSansFallbackFull.ttf` is present on the disk.

I've opened this GitHub issue from user experience perspective.

> The language must be set in minetest.conf

If above is a must, then my question would be, could this be done differently (on Linux or actually any OS)?

Operating system is installed and default language is configured. On Linux via `locale` and appropriate environment variables. Why additional settings in the config? If language is not supported, (not sure is this the case here) like Polish in above example, if there will be no fonts, then print a warning on terminal to stderr and fallback to default language, `en` in this case, so there are some fonts displayed.

Don't you agree that this is strange user experience of no fonts, because `language = en` is missing in config file? Why not make it default then?

**Comment by neoh4x0r** [comment]
> @neoh4x0r, the only thing I've changed, after seeing no fonts and a bit of Google is this:
> 
> ```
>   echo 'language = en' >> ~/.minetest/minetest.conf
> ```
> 
> I didn't need to modify any font settings. I can confirm that `DroidSansFallbackFull.ttf` is present on the disk.
> 
> I've opened this GitHub issue from user experience perspective.
> 
> > The language must be set in minetest.conf
> 
> If above is a must, then my question would be, could this be done differently (on Linux or actually any OS)?
> 
> Operating system is installed and default language is configured. On Linux via `locale` and appropriate environment variables. Why additional settings in the config? If language is not supported, (not sure is this the case here) like Polish in above example, if there will be no fonts, then print a warning on terminal to stderr and fallback to default language, `en` in this case, so there are some fonts displayed.
> 
> Don't you agree that this is strange user experience of no fonts, because `language = en` is missing in config file? Why not make it default then?

What happens you do this ? (does it still work? Do you get translated text?
```
 echo 'language = pl' >> ~/.minetest/minetest.conf
```

**Comment by kucharskim** [comment]
(please bare with me, I don't have access to this particular system at the moment, will try to figure something out)

**Comment by rubenwardy** [comment]
> The language must be set in minetest.conf

This is wrong, Minetest uses the system locale if the language isn't explicitly specified

**Comment by neoh4x0r** [comment]
Just did some testing --

Seems that cmake defaulted the **STATIC_LOCALEDIR** to /usr/local/share/locale
and minetest is looking for /usr/local/share/minetest/locale

```
$ which minetest 
/usr/local/bin/minetest

$ minetest 
initializePaths : path_locale = /usr/local/share/minetest/locale

$ minetest --version
Minetest 5.5.0-dev-debug-c47313db6-dirty (Linux)
Using Irrlicht 1.9.0mt2
Using Lua 5.1.5
BUILD_TYPE=Debug
RUN_IN_PLACE=0
USE_CURL=1
USE_GETTEXT=1
USE_SOUND=1
USE_FREETYPE=1
STATIC_SHAREDIR="/usr/local/share/minetest"
STATIC_LOCALEDIR="/usr/local/share/locale"
```

**Comment by neoh4x0r** [comment]
@kucharskim
I didn't have missing menu text, but I also didn't get any translated menu text when setting the langauge.
(I tested on 5.1.1 and 5.5.0)

On your system what is the output of the following commands:

1. localectl list-locales | grep -i pl
2. apt-cache policy locales-all

I didn't have **locales-all** installed, once it was installed I was getting translated menu text.

**Comment by kucharskim** [comment]
I have remote access to the machine at the moment:

```
# ls -lh /usr/share/games/minetest/locale
ls: cannot access '/usr/share/games/minetest/locale': No such file or directory

# localectl list-locales | grep -i pl
pl_PL.UTF-8

# apt-cache policy locales-all
locales-all:
  Installed: (none)
  Candidate: 2.31-0ubuntu9.2
  Version table:
     2.31-0ubuntu9.2 500
        500 http://pl.archive.ubuntu.com/ubuntu focal-updates/main amd64 Packages
     2.31-0ubuntu9 500
        500 http://pl.archive.ubuntu.com/ubuntu focal/main amd64 Packages

# dpkg -l | grep -cw locales-all
0

# apt install -s locales-all
Reading package lists... Done
Building dependency tree       
Reading state information... Done
The following NEW packages will be installed:
  locales-all
0 upgraded, 1 newly installed, 0 to remove and 16 not upgraded.
Inst locales-all (2.31-0ubuntu9.2 Ubuntu:20.04/focal-updates [amd64])
Conf locales-all (2.31-0ubuntu9.2 Ubuntu:20.04/focal-updates [amd64])
```

Visually will check when will visit that city or find spare machine to replicate installation, from that remote location.

**Comment by neoh4x0r** [comment]
I see **locales-all** was not installed -- this package installs Precompiled locale data (such as the following).
```
/usr/lib/locale/pl_PL.utf8/LC_ADDRESS
/usr/lib/locale/pl_PL.utf8/LC_COLLATE
/usr/lib/locale/pl_PL.utf8/LC_CTYPE
/usr/lib/locale/pl_PL.utf8/LC_IDENTIFICATION
/usr/lib/locale/pl_PL.utf8/LC_MEASUREMENT
/usr/lib/locale/pl_PL.utf8/LC_MESSAGES/SYS_LC_MESSAGES
/usr/lib/locale/pl_PL.utf8/LC_MONETARY
/usr/lib/locale/pl_PL.utf8/LC_NAME
/usr/lib/locale/pl_PL.utf8/LC_NUMERIC
/usr/lib/locale/pl_PL.utf8/LC_PAPER
/usr/lib/locale/pl_PL.utf8/LC_TELEPHONE
/usr/lib/locale/pl_PL.utf8/LC_TIME
```

**Comment by ResuUman** [comment]
No fonts in launcher was fixed here (in 5.2.0):
[fallback font not required](https://github.com/minetest/minetest/commit/391432eb9e318aa386be1a1ca58e5064c135f100#diff-993a190e759d0bff7600ea32c5c12a3ee4cceed9eec0f7e91f0e6c660dbb3266)
(fallback font setting)

This bug is duplicate.
Screenshot in #4152 is a different result of the same bug.
[Bug was introduced in 0.4.14](https://github.com/minetest/minetest/commit/c0e2bc30e5b6adab8fab7b03326d462120cd7707#diff-993a190e759d0bff7600ea32c5c12a3ee4cceed9eec0f7e91f0e6c660dbb3266)

@neoh4x0r
IMHO you are writing about different issue (probably: locales not found) and expect support from @kucharskim
